docgen/src/guide/React_native.md

```diff
@@ -6,9 +6,643 @@ category: guide
+          <View style={{ flexDirection: 'row', alignItems: 'center' }}>
+            <Image
+              style={{ height: 100, width: 100 }}
+              source={{ uri: item.image }}
+            />
+            <View style={{ flex: 1 }}>
+              <Text>
+                {item.name}
+              </Text>
+              <Text>
+                {item.type}
+              </Text>
+            </View>
+          </View>
+        );
+      }}
+    />
+  );
+});
+```
+
+We can then call this component in the render of App:
+
+```jsx
+const App = () =>
+  <InstantSearch
+    appId="latency"
+    apiKey="3d9875e51fbd20c7754e65422f7ce5e1"
+    indexName="bestbuy"
+  >
+   <Hits/>
+  </InstantSearch>
+```
+
+You should now be able to see the results and scroll to fetch more of them. Of course depending of the attributes available on your records you might want to change this template. 
+
+In this section we've seen:
+- how to display the results from Algolia
+- how to build an infinite scrolling experience
+
+## Add a SearchBox
+
+Now that we've added the results, we can start querying our index. To do this, we are gonna use the [connectSearchBox](connectors/connectSearchBox.html) connector. 
+
+This connector gives you two interesting properties: 
+
+* `currentRefinement`, the current query
+* `refine`, the function to call when a new query is entered
+
+On the React Native side we are going to take advantage of the [TextInput](http://facebook.github.io/react-native/releases/0.45/docs/textinput.html#textinput) to render this search box. 
+
+```jsx
+// We need to add the connectSearchBox to our import
+import {
+  connectInfiniteHits,
+  connectSearchBox,
+} from 'react-instantsearch/connectors';
+// We need to add the TextInput to our import
+import {
+  StyleSheet,
+  View,
+  FlatList,
+  Image,
+  Text,
+  TextInput,
+} from 'react-native';
+// [...]
+
+const SearchBox = connectSearchBox(({ refine, currentRefinement }) => {
+  const styles = {
+    height: 60,
+    borderWidth: 1,
+    padding: 10,
+    margin: 10,
+    flex: 1,
+  };
+  return (
+    <TextInput
+      style={styles}
+      onChangeText={text => refine(text)}
+      value={currentRefinement}
+      placeholder={'Search a product...'}
+      clearButtonMode={'always'}
+      spellCheck={false}
+      autoCorrect={false}
+      autoCapitalize={'none'}
+    />
+  );
+});
+
+```
+
+Let’s add this new `<SearchBox>` component to our InstantSearch instance. 
+
+```jsx
+const App = () =>
+   <View style={styles.container}>
+        <InstantSearch
+            appId="latency"
+            apiKey="6be0576ff61c053d5f9a3225e2a90f76"
+            indexName="ikea"
+        >
+            <View
+            style={{
+                flexDirection: 'row',
+            }}
+            >
+            <SearchBox />
+            </View>
+            <Hits />
+        </InstantSearch>
+    </View>
+```
+
+
+The search is now interactive but we don't see what matched in each of the products.
```

Comment:
I suggest to stop the section here to keep them short and sweet. Then use the "but we don't see what matched in each of the products." as a transition to a new section called "Highlighting". At least that's what we did on the mobile side for both iOS and Android